test/e2e/node_feature_discovery_test.go

```diff
@@ -909,6 +909,64 @@ denyLabelNs: []
 					)).NotTo(HaveOccurred())
 				})
 			})
+
+			Context("and test whether resyncPeriod is passed successfully or not", func() {
+				BeforeEach(func(ctx context.Context) {
+					extraMasterPodSpecOpts = []testpod.SpecOption{
+						testpod.SpecWithConfigMap("nfd-master-conf", "/etc/kubernetes/node-feature-discovery"),
+					}
+					cm := testutils.NewConfigMap("nfd-master-conf", "nfd-master.conf", `
+resyncPeriod: "1s"
+`)
+					_, err := f.ClientSet.CoreV1().ConfigMaps(f.Namespace.Name).Create(ctx, cm, metav1.CreateOptions{})
+					Expect(err).NotTo(HaveOccurred())
+				})
+				It("labels should be restored to the original ones", func(ctx context.Context) {
+					// deploy node feature object
+					if !useNodeFeatureApi {
+						Skip("NodeFeature API not enabled")
+					}
+
+					nodes, err := getNonControlPlaneNodes(ctx, f.ClientSet)
+					Expect(err).NotTo(HaveOccurred())
+
+					targetNodeName := nodes[0].Name
+					Expect(targetNodeName).ToNot(BeEmpty(), "No suitable worker node found")
+
+					// Apply Node Feature object
+					By("Creating NodeFeature object")
+					nodeFeatures, err := testutils.CreateOrUpdateNodeFeaturesFromFile(ctx, nfdClient, "nodefeature-1.yaml", f.Namespace.Name, targetNodeName)
+					Expect(err).NotTo(HaveOccurred())
+
+					By("Verifying node labels from NodeFeature object #1")
+					expectedLabels := map[string]k8sLabels{
+						targetNodeName: {
+							nfdv1alpha1.FeatureLabelNs + "/e2e-nodefeature-test-1": "obj-1",
+							nfdv1alpha1.FeatureLabelNs + "/e2e-nodefeature-test-2": "obj-1",
+							nfdv1alpha1.FeatureLabelNs + "/fake-fakefeature3":      "overridden",
+						},
+					}
+					Expect(checkForNodeLabels(ctx, f.ClientSet,
+						expectedLabels, nodes,
+					)).NotTo(HaveOccurred())
+
+					nodeToUpdate := getNode(nodes, targetNodeName)
+					nodeToUpdate.Labels = map[string]string{}
```

Comment:
We don't want to delete all labels from the node. Only delete/change those that NFD created. Actually, you don't need to modify the `nodeToUpdate` at all, it's the "original state" of the node so you could just update to this state

Reply:
Ach, test fails with `the object has been modified; please apply your changes to the latest version and try again` 🧐 Safest would be patch the object, e.g. modifying some of the created labels.

Anyway, this test is a great addition, not only verifying resync period option but also verifying that nfd actually "fixes" tampered nodes

Reply:
the test is actually running in my Minikube cluster... anyway I'll do it the nodes update with the patch